engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsAtomNode.java

```diff
@@ -100,7 +101,8 @@ boolean equalsAtomsWithCustomComparator(
     var args = new Object[] {cachedComparator, self, other};
     var result = invokeNode.execute(compareFn, null, State.create(ctx), args);
     assert orderingOrNullOrError(this, ctx, result, compareFn);
-    return ctx.getBuiltins().ordering().newEqual() == result;
+    var resultWithWarningRemoved = result instanceof WithWarnings ww ? ww.getValue() : result;
```

Comment:
Added a WarningsLibrary parameter -- please review these changes again since this is unfamiliar to me.

Reply:
I ran the benchmarks twice, because there were a few in the first run that were worse, and I wanted to see if they really were worse.

@JaroslavTulach How do these look to you?

Most of the results were like this:
<img width="275" alt="Screen Shot 2024-08-05 at 10 52 41 AM" src="https://github.com/user-attachments/assets/6ce5228f-2c44-449d-bb38-d9d22a5e714b">
<img width="338" alt="Screen Shot 2024-08-05 at 10 53 22 AM" src="https://github.com/user-attachments/assets/8b3ce630-fbf7-4bbe-a8a3-fbef8c641c54">

A few were like this:
<img width="1305" alt="Screen Shot 2024-08-06 at 9 07 15 AM" src="https://github.com/user-attachments/assets/495d15ba-e9be-48c5-a652-ca9b1bcba5a7">
<img width="1390" alt="Screen Shot 2024-08-06 at 9 07 01 AM" src="https://github.com/user-attachments/assets/7fd4bea7-45af-4562-8969-9ce19391a32d">
<img width="1316" alt="Screen Shot 2024-08-06 at 9 06 46 AM" src="https://github.com/user-attachments/assets/d9332880-c946-4eee-9fca-86d3a20bf90b">

Reply:
I think these results are fine.